clang/test/CodeGenSYCL/intel-fpga-no-global-work-offset.cpp

```diff
@@ -1,28 +1,41 @@
-// RUN: %clang_cc1 -fsycl -fsycl-is-device -triple spir64-unknown-unknown-sycldevice -disable-llvm-passes -emit-llvm -o - %s | FileCheck %s
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -internal-isystem %S/Inputs -triple spir64-unknown-unknown-sycldevice -disable-llvm-passes -emit-llvm -o - %s | FileCheck %s
+
+#include "sycl.hpp"
+
+using namespace cl::sycl;
+queue q;
 
 class Foo {
 public:
   [[intel::no_global_work_offset(1)]] void operator()() const {}
 };
 
-template <typename name, typename Func>
-__attribute__((sycl_kernel)) void kernel(const Func &kernelFunc) {
-  kernelFunc();
-}
+template <int SIZE>
+class Functor {
+public:
+  [[intel::no_global_work_offset(SIZE)]] void operator()() const {}
```

Comment:
You should also add a test like:
```
template <int N>
[[intel::no_global_work_offset(N)]] void func() {}
```
as both a member function of this class template and as a free function.

Reply:
Done. Added the test.